x/campaign/keeper/msg_server_add_vesting_options.go

```diff
@@ -19,12 +19,19 @@ func (k msgServer) AddVestingOptions(goCtx context.Context, msg *types.MsgAddVes
 		return nil, sdkerrors.Wrapf(types.ErrCampaignNotFound, "%d", msg.CampaignID)
 	}
 
-	// Get the coordinator ID
-	coordinatorID, found := k.profileKeeper.CoordinatorIDFromAddress(ctx, msg.Coordinator)
+	// Get the coordinator ID associated to the sender address
+	coordByAddress, found := k.profileKeeper.GetCoordinatorByAddress(ctx, msg.Coordinator)
 	if !found {
 		return nil, sdkerrors.Wrap(profiletypes.ErrCoordAddressNotFound, msg.Coordinator)
 	}
-	if campaign.CoordinatorID != coordinatorID {
+
+	coord, _ := k.profileKeeper.GetCoordinator(ctx, coordByAddress.CoordinatorID)
+	if !coord.Active {
+		return nil, sdkerrors.Wrapf(profiletypes.ErrCoordInactive,
+			"coordinator %d inactive", coord.CoordinatorID)
+	}
```

Comment:
We should have this logic in many place since many operation must be done by the coordinator

I think it could be relevant to implement a `GetActiveCoordinatorByAddress` method for simplicity

Then we can add test cases to this method and we don't need to update current tests for messages since the retrieve validator must be active

Reply:
Or maybe even a simpler implement would be to keep `GetCoordinatorByAddress` and return a critical error inside this method if the retrieve a inactive validator since we should unmatch the address in `CoordinatorByAddress` once the coordinator is disabled

And then we could skip
```
coord, _ := k.profileKeeper.GetCoordinator(ctx, coordByAddress.CoordinatorID)
if !coord.Active {
return nil, sdkerrors.Wrapf(profiletypes.ErrCoordInactive,
	"coordinator %d inactive", coord.CoordinatorID)
}
```